services/src/main/groovy/org/jfrog/artifactory/client/impl/UploadableArtifactImpl.groovy

```diff
@@ -61,10 +61,17 @@ class UploadableArtifactImpl extends ArtifactBase<UploadableArtifact> implements
         }
     }
 
-    @Override
     String doUploadAndExplode() {
+        return this.doUploadAndExplode(false)
+    }
+
+    @Override
+    String doUploadAndExplode(boolean atomic) {
         def params = parseParams(props, "=")
         Map<String, String>  headers = ['X-Explode-Archive': true]
+        if(atomic){
+            headers.put('X-Explode-Archive-Atomic',atomic as String)
```

Comment:
The above line can ve probably simplified to:
            headers.put('X-Explode-Archive-Atomic', 'true')